### PR TITLE
Adding missing functionality for Key Transactions

### PR DESCRIPTION
**Pull request by wmak**
- Return nothing when there's no key transactions
   - Right now the endpoint is still querying but with no conditions
   - This change prevents us from querying at all when there's no key transactions
- Adding a `is-key-transactions` endpoint to check if a transaction/project pair is key or not
- Allowing the query param to be passed to key transactions
- TODO: In another PR i'll introduce the ability to hit event-stats and get key transactions back.

### CODE REVIEW

File: src/sentry/discover/endpoints/discover_key_transactions.py [hunk 1]
```diff
@@ -56,9 +68,12 @@ def get(self, request, organization):
 
         queryset = KeyTransaction.objects.filter(organization=organization, owner=request.user)
 
+        if queryset.count() == 0:
```
Comment (dashed) [inline src/sentry/discover/endpoints/discover_key_transactions.py]:
```suggestion
        if not queryset.exists():
```

is this better?

Comment (wmak) [reply src/sentry/discover/endpoints/discover_key_transactions.py]: Yeah, good catch. Exists is more efficient than count

File: src/sentry/discover/endpoints/discover_key_transactions.py [hunk 2]
```diff
@@ -2,29 +2,41 @@
...
+        try:
+            KeyTransaction.objects.get(
+                organization=organization,
+                owner=request.user,
+                project=project,
+                transaction=transaction,
+            )
+            return Response({"isKey": True}, status=200)
+        except KeyTransaction.DoesNotExist:
+            raise ResourceDoesNotExist
```
Comment (wmak) [inline src/sentry/discover/endpoints/discover_key_transactions.py]: @dashed dunno if this is more annoying for the frontend, can change this to `isKey: False` if its better for you.

File: src/sentry/discover/endpoints/discover_key_transactions.py [hunk 3]
```diff
@@ -56,9 +68,12 @@ def get(self, request, organization):
 
         queryset = KeyTransaction.objects.filter(organization=organization, owner=request.user)
 
+        if not queryset.exists():
+            return Response(status=204)
```
Comment (markstory) [inline src/sentry/discover/endpoints/discover_key_transactions.py]: Should this be a 404? as no results could be found.

File: src/sentry/discover/endpoints/discover_key_transactions.py (hunk 2 above)
Comment (dashed) [reply src/sentry/discover/endpoints/discover_key_transactions.py]: yea that might be better.

**Review by dashed — APPROVED** [review]
Looks good to me 👍

**Review by markstory — APPROVED** [review]

**Pull request merged**